### PR TITLE
CATS-XML IO

### PR DESCRIPTION
**Pull request by jzwar**
# Overview
Support CATS-type xml format.

## Showcase
```
import splinepy as sp
cone = sp.helpme.create.cone(5,10,angle=180)
sp.io.cats.export("test.xml", cone)
list_of_splines = sp.io.cats.load("test.xml")
```

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [x] Added test(s)

### CODE REVIEW

File: splinepy/io/cats.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+                )
+            n_patches = int(
+                child.attrib.get(CATS_XML_KEY_WORDS["n_patches"], 0)
+            )
+            for patch_element in child:
+                list_of_splines.append(_read_spline(patch_element))
+        else:
+            debug(f"Unused xml-keyword {child.tag}")
+    debug(f"Found a total of {len(list_of_splines)} " f"BSplines and NURBS")
+    if len(list_of_splines) != n_patches:
```
Comment (j042) [inline splinepy/io/cats.py:160]:
```
    if len(list_of_splines) != n_patches:
UnboundLocalError: local variable 'n_patches' referenced before assignment
```
I guess it doesn't reach l.152 ?

**Comment by j042** [comment]
Have you tried a round-trip? load fails

**Comment by jzwar** [comment]
> Have you tried a round-trip? load fails
There is one in the test. However, I realized, that the SplineLib example does not work because of a default root setting. I updated the code accordingly and now everything should be fine.

File: splinepy/io/cats.py (hunk 1 above)
Comment (jzwar) [reply splinepy/io/cats.py:159]: Previously root was "xml" however, it is now updated to `SplineList`

**Review by j042 — APPROVED** [review]
thank you so much!

**Pull request merged**